File: app/src/main/java/com/shop440/Adapters/CategorySearchAdapter.java
package com.shop440.Adapters;

import android.content.Context;
import android.content.Intent;
import android.graphics.Typeface;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.shop440.models.CategoryModel;
import com.shop440.R;
import com.shop440.SearchResultActivity;

import java.util.ArrayList;

/**
 * Created by SMILECS on 2/7/17.
 */

public class CategorySearchAdapter extends RecyclerView.Adapter<CategorySearchAdapter.ViewHolder>{
    ArrayList<CategoryModel> model;
    Context c;

    public CategorySearchAdapter(Context c, ArrayList<CategoryModel> model){
        this.c = c;
        this.model = model;
    }

    public static class ViewHolder extends RecyclerView.ViewHolder{
        TextView category;
        public ViewHolder(final View itemView) {
            super(itemView);
            category = (TextView) itemView.findViewById(R.id.category);
            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    CategoryModel toPass = (CategoryModel) itemView.getTag();
                    Intent i = new Intent(view.getContext(), SearchResultActivity.class);
                    i.putExtra("query", toPass.getSlug());
                    i.putExtra("title", toPass.getName());
                    i.putExtra("isSearch", false);
                    view.getContext().startActivity(i);
                }
            });
        }
    }

    @Override
    public CategorySearchAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.category_view, parent, false);
        CategorySearchAdapter.ViewHolder vh = new CategorySearchAdapter.ViewHolder(v);
        return vh;
    }

    @Override
    public void onBindViewHolder(CategorySearchAdapter.ViewHolder holder, int position) {
        Typeface robotCondensed = Typeface.createFromAsset(c.getAssets(),
                "fonts/Roboto-Thin.ttf");
        CategoryModel categoryModel = model.get(position);
        holder.category.setTypeface(robotCondensed);
        holder.category.setText(categoryModel.getName());
        holder.itemView.setTag(categoryModel);
        //holder.category.setTag(md);
    }

    @Override
    public int getItemCount() {
        return model.size();
    }
}

File: app/src/main/java/com/shop440/adapters/CategorySearchAdapter.kt
package com.shop440.adapters


import android.content.Context
import android.graphics.Typeface
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.shop440.R
import com.shop440.dao.models.CategoryModel
import java.util.*

/**
 * Created by SMILECS on 2/7/17.
 */

class CategorySearchAdapter(val c: Context, val model: ArrayList<CategoryModel>) : RecyclerView.Adapter<CategorySearchAdapter.ViewHolder>() {
    lateinit var categoryQuery:CategoryQuery
    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var category: TextView = itemView.findViewById(R.id.category) as TextView
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CategorySearchAdapter.ViewHolder {
        val v = LayoutInflater.from(parent.context).inflate(R.layout.category_view, parent, false)
        return ViewHolder(v)
    }

    override fun onBindViewHolder(holder: CategorySearchAdapter.ViewHolder, position: Int) {
        val robotCondensed = Typeface.createFromAsset(c.assets,
                "fonts/Roboto-Thin.ttf")
        val categoryModel = model[position]
        holder.category.typeface = robotCondensed
        holder.category.text = categoryModel.catName
        holder.itemView.tag = categoryModel
        holder.itemView.setOnClickListener { _ ->
            categoryQuery.onCategorySelected(categoryModel.slug)
        }
    }

    override fun getItemCount(): Int {
        return model.size
    }

    interface CategoryQuery{
        fun onCategorySelected(cat:String)
    }
}